src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/step/VariantLoaderStepParametersValidatorTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.parameters.validation.step;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersInvalidException;
+import org.springframework.batch.core.JobParametersValidator;
+
+import uk.ac.ebi.eva.pipeline.parameters.JobParametersNames;
+import uk.ac.ebi.eva.test.rules.PipelineTemporaryFolderRule;
+import uk.ac.ebi.eva.test.utils.TestFileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Tests that the arguments necessary to run a {@link uk.ac.ebi.eva.pipeline.jobs.steps.VariantLoaderStep} are correctly
+ * validated
+ */
+public class VariantLoaderStepParametersValidatorTest {
+
+    private JobParametersValidator validator;
+
+    @Rule
+    public PipelineTemporaryFolderRule temporaryFolderRule = new PipelineTemporaryFolderRule();
+
+    private Map<String, JobParameter> parameters;
+
+    private Map<String, JobParameter> optionalParameters;
+
+    @Before
+    public void setUp() throws IOException {
+        validator = new VariantLoaderStepParametersValidator();
+        final File dir = TestFileUtils.getResource("/parameters-validation/");
+        final File inputVcf = TestFileUtils.getResource("/parameters-validation/file.vcf.gz");
```

Comment:
Better to use `temporaryFolderRule` and `temporaryFolderRule.newFile()`, so we don't need to set the flags to true. It will also prevent interactions among concurrent tests, where some could set them to false and others to true. Finally, if all the validations are run on temporary folders then we don't need the `parameters-validation` folder.

Also applies to `VepAnnotationGeneratorStepParametersValidatorTest`.